### PR TITLE
Implement remove items from cart

### PR DESCRIPTION
**Pull request by quwee**
Created handler for remove from cart requests, service method RemoveItem and repository methods ExistCart, ExistItem and RemoveItem.

### CODE REVIEW

File: internal/service/service.go [hunk 1]
```diff
@@ -28,3 +40,55 @@ func (s *CartService) Create() (int, error) {
...
+	const op = "service.RemoveItem"
+
+	s.log.Debug(op, "cartID", cartID, "itemID", itemID)
+
+	err := checkCartExists(s, cartID)
+	if err != nil {
+		return err
+	}
+
+	err = checkItemExists(s, itemID)
```
Comment (stassanakoyeu) [inline internal/service/service.go:54]: maybe it will be better to make it private method?

File: internal/service/service.go [hunk 2]
```diff
@@ -28,3 +40,55 @@ func (s *CartService) Create() (int, error) {
...
+	return nil
+}
+
+func checkItemExists(s *CartService, itemID int) error {
+	const op = "service.checkItemExists"
+
+	s.log.Debug(op, "itemID", itemID)
+
+	exists, err := s.r.ExistItem(itemID)
+	switch {
```
Comment (stassanakoyeu) [inline internal/service/service.go:87]: i think that for 2 cases there is no sense to make switch, let's just do 2 if

File: internal/service/service.go [hunk 3]
```diff
@@ -28,3 +40,55 @@ func (s *CartService) Create() (int, error) {
...
+}
+
+func checkCartExists(s *CartService, cartID int) error {
+	const op = "service.checkCartExists"
+
+	s.log.Debug(op, "cartID", cartID)
+
+	exists, err := s.r.ExistCart(cartID)
+	switch {
+	case err != nil:
```
Comment (stassanakoyeu) [inline internal/service/service.go:73]: same

**Review by stassanakoyeu — CHANGES_REQUESTED** [review]

File: internal/service/service.go (hunk 1 above)
Comment (quwee) [reply internal/service/service.go:54]: isn't it already private?

**Pull request merged**